include/sparrow_ipc/arrow_interface/arrow_array/private_data.hpp

```diff
@@ -1,25 +1,46 @@
 #pragma once
-
+#include <concepts>
 #include <cstdint>
 #include <vector>
 
 #include "sparrow_ipc/config/config.hpp"
 
 namespace sparrow_ipc
 {
+    template <typename T>
+    concept ArrowPrivateData = requires(T& t)
+    {
+        { t.buffers_ptrs() } -> std::same_as<const void**>;
+        { t.n_buffers() } -> std::convertible_to<std::size_t>;
+    };
+
+    class owning_arrow_array_private_data
```

Comment:
Hmm, but if you want to efficiently handle the case where some buffers are compressed and others not, you may want a more flexible facility, e.g.:
```c++
class arrow_array_private_data
{
public:
    using optionally_owned_buffer = std::variant<std::span<const uint8_t>, std::vector<uint8_t>>;
    explicit arrow_array_private_data(std::vector<optionally_owned_buffer>&& buffers);
};
```

Reply:
(in the longer term, sparrow might want a reference-counted optionally-owning buffer class...)